ReactWindows/ReactNative/Animated/SpringAnimationDriver.cs

```diff
@@ -0,0 +1,245 @@
+                // springs in one pass of the BaseSpringSystem.
+
+                // The initial derivative is based on the current velocity and the calculated acceleration
+                aVelocity = velocity;
+                aAcceleration = (tension * (_endValue - tempPosition)) - friction * velocity;
+
+                // Calculate the next derivatives starting with the last derivative and integrating over the
+                // timestep
+                tempPosition = position + aVelocity * SolverTimestepSec * 0.5;
+                tempVelocity = velocity + aAcceleration * SolverTimestepSec * 0.5;
+                bVelocity = tempVelocity;
+                bAcceleration = (tension * (_endValue - tempPosition)) - friction * tempVelocity;
+
+                tempPosition = position + bVelocity * SolverTimestepSec * 0.5;
+                tempVelocity = velocity + bAcceleration * SolverTimestepSec * 0.5;
+                cVelocity = tempVelocity;
+                cAcceleration = (tension * (_endValue - tempPosition)) - friction * tempVelocity;
+
+                tempPosition = position + cVelocity * SolverTimestepSec;
+                tempVelocity = velocity + cAcceleration * SolverTimestepSec;
+                dVelocity = tempVelocity;
+                dAcceleration = (tension * (_endValue - tempPosition)) - friction * tempVelocity;
+
+                // Take the weighted sum of the 4 derivatives as the final output.
+                dxdt = 1.0 / 6.0 * (aVelocity + 2.0 * (bVelocity + cVelocity) + dVelocity);
+                dvdt = 1.0 / 6.0 * (aAcceleration + 2.0 * (bAcceleration + cAcceleration) + dAcceleration);
+
+                position += dxdt * SolverTimestepSec;
+                velocity += dvdt * SolverTimestepSec;
+            }
+
+            _tempState.Position = tempPosition;
+            _tempState.Velocity = tempVelocity;
+
+            _currentState.Position = position;
+            _currentState.Velocity = velocity;
+
+            if (_timeAccumulator > 0)
+            {
+                Interpolate(_timeAccumulator / SolverTimestepSec);
+            }
+
+            // End the spring immediately if it is overshooting and overshoot clamping is enabled.
+            // Also make sure that if the spring was considered within a resting threshold that it's now
+            // snapped to its end value.
+            if (IsAtRest() || (_overshootClampingEnabled && IsOvershooting()))
+            {
+                // Don't call setCurrentValue because that forces a call to onSpringUpdate
+                if (tension > 0)
+                {
+                    _startValue = _endValue;
+                    _currentState.Position = _endValue;
+                }
+                else
+                {
+                    _endValue = _currentState.Position;
+                    _startValue = _endValue;
+                }
+
+                _currentState.Velocity = 0;
+            }
+        }
+
+        class PhysicsState
```

Comment:
Yeah, I suppose non-readonly structs would work.

---
In reply to: [74450042](https://github.com/ReactWindows/react-native-windows/pull/583#discussion_r74450042) [](ancestors = 74450042)